pkg/services/queryhistory/api.go

```diff
@@ -61,6 +62,10 @@ func (s *QueryHistoryService) createHandler(c *contextmodel.ReqContext) response
 // 401: unauthorisedError
 // 500: internalServerError
 func (s *QueryHistoryService) searchHandler(c *contextmodel.ReqContext) response.Response {
+	if (c.GetOrgRole() == org.RoleViewer && !s.Cfg.ViewersCanEdit) {
```

Comment:
Wanted to note I attempted to fix this by attaching the appropriate middleware permissions in https://github.com/grafana/grafana/blob/main/pkg/services/queryhistory/api.go#L18, but the role is strict and does not consider the viewers_can_edit config.